src/federation_server.js

```diff
@@ -20,6 +20,7 @@ export const FEDERATION_RESPONSE_MAX_SIZE = 100 * 1024;
  * @param {string} domain Domain this server represents
  * @param {object} [opts]
  * @param {boolean} [opts.allowHttp] - Allow connecting to http servers, default: `false`. This must be set to false in production deployments! You can also use {@link Config} class to set this globally.
+ * @param {number} [opts.timeout] - Allow a timeout, default: 0. In ms.
  */
 export class FederationServer {
   constructor(serverURL, domain, opts = {}) {
```

Comment:
Static method instantiate this class but `opts.timeout` is not persisted anywhere.